docs/source/reference/index.rst

```diff
@@ -23,6 +23,9 @@ Chainer
    :toctree: generated/
    :nosignatures:
 
+   optuna.integration.AllenNLPExecutor
+   optuna.integration.allennlp.dump_best_config
+   optuna.integration.AllenNLPPruningCallback
```

Comment:
These modules are listed inside the `Chainer` section. I think we should create the `AllenNLP` section.